contracts/services/neutron-ibc-transfer/src/contract.rs

```diff
@@ -0,0 +1,141 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{to_json_binary, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+use neutron_sdk::bindings::query::NeutronQuery;
+use valence_service_utils::{
+    error::ServiceError,
+    msg::{ExecuteMsg, InstantiateMsg},
+};
+
+use crate::msg::{ActionMsgs, Config, QueryMsg, ServiceConfig, ServiceConfigUpdate};
+
+// version info for migration info
+const CONTRACT_NAME: &str = env!("CARGO_PKG_NAME");
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut<NeutronQuery>,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg<ServiceConfig>,
+) -> Result<Response, ServiceError> {
+    valence_service_base::instantiate(deps.into_empty(), CONTRACT_NAME, CONTRACT_VERSION, msg)
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut<NeutronQuery>,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg<ActionMsgs, ServiceConfigUpdate>,
+) -> Result<Response, ServiceError> {
+    valence_service_base::execute(
+        deps,
+        env,
+        info,
+        msg,
+        actions::process_action,
+        execute::update_config,
+    )
+}
+
+mod actions {
+    use cosmwasm_std::{DepsMut, Env, MessageInfo, Response};
+    use neutron_sdk::bindings::query::NeutronQuery;
+    use valence_service_utils::{error::ServiceError, execute_on_behalf_of};
+
+    use crate::msg::{ActionMsgs, Config, IbcTransferAmount};
+
+    pub fn process_action(
+        deps: DepsMut<NeutronQuery>,
+        env: Env,
+        _info: MessageInfo,
+        msg: ActionMsgs,
+        cfg: Config,
+    ) -> Result<Response, ServiceError> {
+        match msg {
+            ActionMsgs::IbcTransfer {} => {
+                let balance = cfg.denom().query_balance(&deps.querier, cfg.input_addr())?;
+
+                let amount = match cfg.amount() {
+                    IbcTransferAmount::FullAmount => balance,
+                    IbcTransferAmount::FixedAmount(amount) => {
+                        if balance < *amount {
+                            return Err(ServiceError::ExecutionError(format!(
+                                "Insufficient balance for denom '{}' in config (required: {}, available: {}).",
+                                cfg.denom(), amount, balance,
+                            )));
+                        }
+                        *amount
+                    }
+                };
+
+                // IBC Transfer funds from input account to output account on the remote chain
+                let block_time = env.block.time;
+                let ibc_send_msg = valence_ibc_utils::neutron::ibc_send_message(
+                    deps,
+                    env,
+                    cfg.remote_chain_info().channel_id.clone(),
+                    cfg.remote_chain_info().port_id.clone(),
```

Comment:
The way I see it, and extending on above, anything that is not "transfer" port here should not work as Neutron MsgTransfer is an IBC transfer would only work with port "transfer"